### ISSUE TITLE
When unpickling from Tasty, we cannot distinguish between overloaded alternatives with the same signature

### ISSUE THREAD

**Issue by smarter** [lampepfl/dotty#7022]
This code compiles fine:
```scala
trait P[A] {
  def foo[T](x: Int): A = ???
}

class C extends P[Int] {
  def foo(x: Int): Int = x
}

object Test {
  def test(p: C): Unit = {
    p.foo(1)
  }
}
```
But if we run it through the pickler/unpickler (easiest way is to do `dotc -Ythrough-tasty Test.scala`), we get:
```scala
-- Error: /home/smarter/opt/dotty/try/tt.scala:9:4 -----------------------------
9 |    p.foo(1)
  |    ^^^^^^^
  |    overloaded alternatives of method foo in class C with types
  |     (x: Int): Int
  |     [T](x: Int): Int does not take parameters
```
The problem is that in Tasty, we use this representation:
```scala
SELECT 32 [foo[Signed Signature(List(scala.Int),scala.Int)]]
```
This uses the denotation signature and not the symbol signature, so both alternatives have the same signature.
Fixing this is necessary to switch to the 2.13 collections.

**Comment by smarter** [comment]
Possible fixes:
- store the symbol signature instead of the denotation signature in tasty and use that to choose the alternative, not sure if that would always work
- store the full type of the alternative in tasty if we detect that the signature is not unique

/cc @odersky

**Comment by odersky** [comment]
@smarter Is this one fixed now?

**Comment by smarter** [comment]
Yes, fixed by f95153437ef989810b3095aa7d0aa789e620da15
